### ISSUE TITLE
Implement hyperparameter tuner with training objective

### ISSUE THREAD

**Issue by Acusick1** [Acusick1/odds#81]
**Goal**: Create hyperparameter tuning system with Optuna, including objective function for training optimization.

**Complexity**: 🟡 Medium

**Location**:
- `packages/odds-analytics/odds_analytics/training/tuning.py` (new file)

**Implementation approach**:
- Create `HyperparameterTuner` ABC with objective creation built-in
- Implement `OptunaTuner` using Optuna v3.x patterns
- Objective function wraps existing `train_with_cv()` for XGBoost, `train_from_config()` for LSTM
- Use PostgreSQL storage for study persistence (consistent with existing infra)

**Scope**:

1. **HyperparameterTuner ABC**:
   - `create_study(study_name, direction, sampler, pruner, storage)` - initialize optimization
   - `optimize(objective_fn, n_trials, timeout, callbacks)` - run optimization
   - `get_best_params()` / `get_best_value()` - retrieve results
   - `get_trials_dataframe()` - export for analysis

2. **OptunaTuner Implementation**:
   - Map `SearchSpace` config to Optuna suggest methods:
     - `type: int` → `trial.suggest_int(name, low, high, step=step)`
     - `type: float` → `trial.suggest_float(name, low, high, log=log)`
     - `type: categorical` → `trial.suggest_categorical(name, choices)`
   - Support samplers: TPE (default), Random, CMA-ES
   - Support pruners: Median, Hyperband, None
   - **Storage**: PostgreSQL URL from existing `DATABASE_URL` config

3. **Objective Function Factory** (integrated):
   ```python
   def create_objective(
       strategy: XGBoostLineMovementStrategy | LSTMLineMovementStrategy,
       config: MLTrainingConfig,
       events: list[Event],  # Raw data, not pre-extracted features
       session: AsyncSession,
   ) -> Callable[[optuna.Trial], float]:
   ```
   - Suggests params from trial using `TuningConfig.search_spaces`
   - **Model params** (e.g., `n_estimators`): Override config directly
   - **Feature params** (e.g., `features.movement_threshold`): Re-extract features
   - Calls `train_with_cv()` (XGBoost) or `train_from_config()` (LSTM)
   - Reports intermediate values for pruning (XGBoost rounds / LSTM epochs)
   - Returns metric specified in `TuningConfig.metric`

   **Feature param handling**:
   - Keys prefixed with `features.` map to FeatureConfig fields
   - If any feature param changes, re-run feature extraction for that trial
   - Cache extracted features by config hash to avoid redundant extraction

4. **Pruning Integration**:
   - XGBoost: Use `xgb.callback.EarlyStopping` + report eval metric per round
   - LSTM: Report validation loss per epoch via `trial.report(val_loss, epoch)`

**Success criteria**:
- [ ] Abstract interface with objective creation built-in
- [ ] OptunaTuner maps SearchSpace to Optuna correctly
- [ ] PostgreSQL storage works for resumable studies
- [ ] Objective correctly trains and returns validation metric
- [ ] Feature param tuning re-extracts features when needed
- [ ] Pruning works for both XGBoost and LSTM
- [ ] Failed trials handled gracefully (logged, study continues)
- [ ] Unit tests cover param suggestion → training → metric flow

**Prerequisites**:
- #71 (Feature extraction params) - ensures model params configurable

**Related files**:
- `packages/odds-analytics/odds_analytics/training/config.py` - `TuningConfig`, `SearchSpace` already defined
- `packages/odds-analytics/odds_analytics/training/__init__.py` - export new classes

**Notes**:
- Consolidates original #81 and #82 - objective function is tightly coupled to tuner
- Search spaces support both model and feature params (see Parameter Scope below)
- Use `optuna.create_study(storage="postgresql://...")` for persistence
- Feature param tuning requires re-extraction per trial (more expensive)
- Optuna v3.x deprecated some v2.x patterns - use current best practices
- Consider `optuna.logging.set_verbosity()` for controlling output

---

**Parameter Scope - Two Categories**:

1. **Model Parameters** (cheap to tune - no re-extraction):
   - XGBoost: n_estimators, max_depth, learning_rate, subsample, reg_alpha, etc.
   - LSTM: hidden_size, num_layers, dropout, epochs, learning_rate, etc.

2. **Feature Parameters** (expensive - requires re-extraction per trial):

   **Safely tunable** (don't change dimensions):
   - `movement_threshold` (0.0-0.1) - trajectory feature sensitivity
   - `opening_tier` / `closing_tier` / `decision_tier` - which snapshots to use
   - `normalize` (bool) - feature preprocessing

   **NOT tunable** (would break model):
   - `markets` - changes feature count
   - `sharp_bookmakers` / `retail_bookmakers` - changes feature count
   - `feature_groups` - incompatible output shapes
   - `lookback_hours` / `timesteps` - changes LSTM tensor shape

**Implementation approach for feature params**:
- SearchSpace keys prefixed with `features.` apply to FeatureConfig
- Tuner detects feature param changes and re-extracts features for that trial
- Cache raw event/odds data, only re-run feature extraction
